src/test/java/org/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,241 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import lombok.SneakyThrows;
+import org.bookstore.dto.book.BookDto;
+import org.bookstore.dto.book.CreateBookRequestDto;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(scripts = {
+        "classpath:/scripts/categories/create-categories.sql",
+        "classpath:/scripts/books/create-books.sql"
+})
+@Sql(scripts = {
+        "classpath:scripts/categories/delete-categories.sql",
+        "classpath:scripts/books/delete-books.sql"},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    private static ObjectMapper objectMapper;
+
+    @BeforeAll
+    @SneakyThrows
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+    }
+
+    @Test
+    @DisplayName("Create book with valid dto")
+    @Sql(scripts = {
+            "classpath:/scripts/books/delete-books.sql",
+            "classpath:/scripts/categories/create-categories.sql"
+    })
+    @WithMockUser(username = "admin", authorities = {"USER", "ADMIN"})
+    void createBook_validRequestDto_returnsDto() throws Exception {
+        CreateBookRequestDto request = new CreateBookRequestDto(
+                "title",
+                "author",
+                "12345678900",
+                BigDecimal.TEN,
+                "descr",
+                "image",
+                Set.of(1L)
+        );
+
+        MvcResult result = mockMvc.perform(post("/books")
+                        .content(objectMapper.writeValueAsBytes(request))
```

Comment:
can't you pass the request object directly without to-byte conversion?

pls double check